internal/clients/kafka/acl/acl.go

```diff
@@ -0,0 +1,194 @@
+func List(ctx context.Context, cl *kadm.Client, accessControlList *AccessControlList) (*AccessControlList, error) {
+
+	o, _ := kmsg.ParseACLOperation(strings.ToLower(accessControlList.AccessControlListOperation))
+	ao := []kadm.ACLOperation{o}
+
+	rpt, _ := kmsg.ParseACLResourcePatternType(strings.ToLower(accessControlList.ResourcePatternTypeFilter))
+
+	b := kadm.ACLBuilder{}
+	ab := b.Topics(accessControlList.Name).Allow(accessControlList.AccessControlListPrinciple).AllowHosts(accessControlList.AccessControlListHost).Operations(ao[0]).ResourcePatternType(rpt)
+
+	resp, err := cl.DescribeACLs(ctx, ab)
+	exists := resp[0].Described
+
+	if exists == nil {
+		return nil, errors.Wrap(err, "cannot describe ACL, it does not exist")
+	}
+
+	acl := AccessControlList{}
+	acl.ResourceType = accessControlList.ResourceType
+	acl.AccessControlListPrinciple = accessControlList.AccessControlListPrinciple
+	acl.AccessControlListHost = accessControlList.AccessControlListHost
+	acl.AccessControlListOperation = accessControlList.AccessControlListOperation
+	acl.AccessControlListPermissionType = accessControlList.AccessControlListPermissionType
+	acl.ResourcePatternTypeFilter = accessControlList.ResourcePatternTypeFilter
+
+	return &acl, nil
+}
+
+// Create creates an ACL from the Kafka side
+func Create(ctx context.Context, cl *kadm.Client, accessControlList *AccessControlList) error {
+
+	o, _ := kmsg.ParseACLOperation(strings.ToLower(accessControlList.AccessControlListOperation))
+	ao := []kadm.ACLOperation{o}
+
+	rpt, _ := kmsg.ParseACLResourcePatternType(strings.ToLower(accessControlList.ResourcePatternTypeFilter))
+
+	switch accessControlList.ResourceType {
+
+	case "Topic":
+
+		b := kadm.ACLBuilder{}
+		ab := b.Topics(accessControlList.Name).Allow(accessControlList.AccessControlListPrinciple).AllowHosts(accessControlList.AccessControlListHost).Operations(ao[0]).ResourcePatternType(rpt)
+
+		resp, err := cl.CreateACLs(ctx, ab)
+		if err != nil {
+			return err
+		}
+
+		a := resp[0].Principal
+		if len(a) == 0 {
+			return errors.New("no create response for acl")
+		}
+
+	case "Group":
+
+		b := kadm.ACLBuilder{}
+		ab := b.Groups(accessControlList.Name).Allow(accessControlList.AccessControlListPrinciple).AllowHosts(accessControlList.AccessControlListHost).Operations(ao[0]).ResourcePatternType(rpt)
+
+		resp, err := cl.CreateACLs(ctx, ab)
+		if err != nil {
+			return err
+		}
+
+		a := resp[0].Principal
+		if len(a) == 0 {
+			return errors.New("no create response for acl")
+		}
+
+	case "TransactionalID":
+
+		b := kadm.ACLBuilder{}
+		ab := b.TransactionalIDs(accessControlList.Name).Allow(accessControlList.AccessControlListPrinciple).AllowHosts(accessControlList.AccessControlListHost).Operations(ao[0]).ResourcePatternType(rpt)
+
+		resp, err := cl.CreateACLs(ctx, ab)
+		if err != nil {
+			return err
+		}
+
+		a := resp[0].Principal
+		if len(a) == 0 {
+			return errors.New("no create response for acl")
+		}
+
+		return nil
+
+	}
+
+	return nil
```

Comment:
Could be refactored as:

```suggestion
	b := kadm.ACLBuilder{}
	ab := b.Allow(accessControlList.AccessControlListPrinciple).AllowHosts(accessControlList.AccessControlListHost).Operations(ao[0]).ResourcePatternType(rpt)

	switch accessControlList.ResourceType {
	case "Topic":
		ab = ab.Topics(accessControlList.Name)
	case "Group":
		ab = ab.Groups(accessControlList.Name)
	case "TransactionalID":
		ab = ab.TransactionalIDs(accessControlList.Name)
	}

	resp, err := cl.CreateACLs(ctx, ab)
	if err != nil {
		return err
	}

	a := resp[0].Principal
	if len(a) == 0 {
		return errors.New("no create response for acl")
	}

	return nil
```